version: "3.9"

services:
  db:
    image: postgres:13
    environment:
      POSTGRES_USER: nutrition_user
      POSTGRES_PASSWORD: nutrition_pass
      POSTGRES_DB: nutrition
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U $${POSTGRES_USER} -d $${POSTGRES_DB}"]
      interval: 5s
      timeout: 5s
      retries: 5
    volumes:
      - dbdata:/var/lib/postgresql/data
      - ./Database:/docker-entrypoint-initdb.d
    ports:
      - "5432:5432"
    networks:
      - nutrition-net

  backend:
    build:
      context: .
      dockerfile: Backend/Dockerfile
    environment:
      FLASK_ENV: development
      DATABASE_URL: postgresql://nutrition_user:nutrition_pass@db:5432/nutrition
    depends_on:
      db:
        condition: service_healthy
    volumes:
      - ./Backend:/app
    command: flask --app backend.py --debug run --host=0.0.0.0
    networks:
      - nutrition-net
    expose:
      - "5000"

  frontend:
    build:
      context: .
      dockerfile: Frontend/nutrition-frontend/Dockerfile
    depends_on:
      - backend
    ports:
      - "3000:80"
    networks:
      - nutrition-net

volumes:
  dbdata:


networks:
  nutrition-net:
